text/0096-metric-instrument-terminology.md

```diff
@@ -0,0 +1,153 @@
+categories of instrument, called "synchronous" vs. "asynchronous",
+named "Measure" and "Observer" in the abstract.  This proposal
+identified four kinds of "refinement" and mapped out the space of
+_possible_ instruments, while not proposing which would actually be
+included in the standard.
+
+[OTEP 93](https://github.com/open-telemetry/oteps/pull/93) followed
+with a list of six standard instruments, the most necessary and useful
+combination of instrument refinements, plus one special case used to
+record timing measurements.
+
+This proposal finalizes the names used to describe the seven
+instruments above, seeking to address core confusion related to
+"Measure":
+
+1. OTEP 88 stipulates that the terms currently in use to name
+synchronous and asynchronous instruments become abstract, but also
+using "Measure-like" and "Observer-like" to discuss instruments with
+refinements.  This proposal states that we shall prefer the
+adjectives, commonly abbreviated "Sync" and "Async", when describing
+instruments.
+2. Prior to OTEP 88, but even with OTEPs 88 and 93 included, there is
+inconsistency in the naming of instruments.  Note that "Counter" and
+"Observer" end in "-er", a noun suffix used in the sense of "[person
+occupationally connected
+with](https://www.merriam-webster.com/dictionary/-er)", while the term
+"Measure" does not fit this pattern.  This proposal proposes to
+replace the abstract term "Measure" by "Recorder", since the
+associated method name (verb) is specified as `Record()`.
+3. The OTEP 93 asynchronous instruments ("LastValueObserver",
+"DeltaObserver", and "CumulativeObserver") have the pattern
+"-Observer", while the OTEP 93 synchronous instruments
+("Counter", "UpDownCounter", "Distribution", "Timing") do not.  This
+proposal keeps "Counter" and "UpDownCounter" for Sum-only synchronous instruments, and does the same
+with "Recorder", yielding "Recorder" and "TimingRecorder".
+4. Confusion over the loss of "Gauge" is addressed by replacing
+"LastValueObserver" with "GaugeObserver".
+
+This proposal also repeats the current specification of the default
+Aggregator for each kind of instrument.
+
+## Explanation
+
+The following table summarizes the four synchronous instruments and
+three asynchronous instruments that will be standardized as a result
+of this set of proposals.
+
+| Existing name | OTEP 93 name       | **Final name**         | Sync or Async | Function | Default aggregation | Rate support |
+| ------------- | ------------------ | ---------------------- | ----------- | ------------- | ---------- | ---- |
+| Counter       | Counter            | **Counter**            | Sync  | Add() | Sum | Yes | 
+|               | UpDownCounter      | **UpDownCounter**      | Sync  | Add() | Sum | Yes |
+| Measure       | Distribution       | **Recorder**           | Sync  | Record() | MinMaxSumCount | No |
+|               | Timing             | **TimingRecorder**     | Sync  | Record() | MinMaxSumCount  | No |
```

Comment:
Does "timing" allow negative values?  What is the difference between Recorder and Timing. I feel that for all the other 6 types we do have them because they express different types of data or time correlation but this is just a helper for "Recorder" that has a specific "unit". I would suggest we focus on the main types first then we add extensions.

Reply:
No, Timing wouldn't allow negatives (as stated in 0093). It is just a helper, but it helps with a mistake that's extremely easy to make. We don't want Java processes reporting milliseconds while Go processes report nanoseconds. Without a conventional instrument, that's going to happen.